lib/spack/spack/ci.py

```diff
@@ -2176,13 +2176,13 @@ def __init__(self, ci_cdash):
     def args(self):
         return [
             "--cdash-upload-url",
-            self.upload_url,
+            f'"{self.upload_url}"',
```

Comment:
It's a combination of spaces and other characters that would otherwise cause the argument parsing to decompose the url into useless gibberish. I can't recall the exact issue off the top of my head but there was more to it than spaces. 
That said, all issues were due to Windows caveats.